File: 1.arrays/max-sliding-window/max_sliding_window.py
def max_sliding_window(arr, k):
  result = []
  if len(arr) <= (k-1):
    result.append(max(arr))
    return result
  maxValue = max(arr[0:k])
  for i in range(k,len(arr)-1):
    current = arr[i]
    print(current)
    if current >= maxValue:
      maxValue = current
    result.append(maxValue)
  return result

arr = [1, 2,5,7,6,8,32,1,2]
k = 3
print(max_sliding_window(arr, k))

File: 1.arrays/max-sliding-window/max_sliding_window.rb
def max_sliding_window(arr, k)
  result = []
  if arr.length <= (k-1)
    result.push(arr.max)
    return result
  end
  maxValue = arr[0..(k-1)].max
  for i in k..(arr.length-1)
    if arr[i] > maxValue
      maxValue = arr[i]
    end
    result.push(maxValue)
  end
  return result
end

arr = [1, 2, 3, 1, 4, 5, 2, 3, 6];
k = 3;
p (max_sliding_window(arr, k));